lib/api_endpoint/api_type_handler.py

```diff
@@ -50,7 +50,8 @@ def visit_generic(
 
             if generic_instantiation.map_value_type.category == 'USER_DEFINED':
                 new_type['additionalProperties'] = {
-                    '$ref': ref_path + generic_instantiation.map_value_type.user_defined_type.resource_id}
+                    '$ref': ref_path + utils.get_str_camel_case(
+                        generic_instantiation.map_value_type.user_defined_type.resource_id, ".", "-", "_")}
```

Comment:
I don't think "-" is needed here, is it? 

This applies to other places in this change. It is worth it to declare a constant somewhere for the word_boundary_characters we recognize.

Reply:
Extracted it as a constant and removed '-'. Some changes to the tests were added (namely adding '-' here and there)